lib/oli_web/controllers/cognito_controller.ex

```diff
@@ -0,0 +1,133 @@
+defmodule OliWeb.CognitoController do
+  use OliWeb, :controller
+
+  import Oli.{HTTP, Utils}
+
+  require Logger
+
+  alias Oli.{Repo, Sections, Accounts, Groups, Institutions}
+  alias Oli.Institutions.SsoJwk
+  alias Oli.Delivery.Sections
+  alias Oli.Delivery.Sections.Section
+
+  @jwks_relative_path "/.well-known/jwks.json"
+
+  def launch(
+        conn,
+        %{
+          "product_id" => product_id,
+          "cognito_id_token" => jwt,
+          "error_url" => _error_url,
+          "community_id" => community_id
+        } = params
+      ) do
+    with {:ok, jwk} <- get_jwk(jwt),
+         {true, %{fields: jwt_fields}, _} <- JOSE.JWT.verify_strict(jwk, ["RS256"], jwt),
+         %Section{} = section <- Sections.get_section_by(slug: product_id),
```

Comment:
We probably want "create lms user" and "create community account" to be atomic operations, therefore this could should be wrapped inside of a transaction